test/tests/components/ws-daemon/cpu_burst_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package wsdaemon
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	csapi "github.com/gitpod-io/gitpod/content-service/api"
+	daemon "github.com/gitpod-io/gitpod/test/pkg/agent/daemon/api"
+	wsapi "github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	wsmanapi "github.com/gitpod-io/gitpod/ws-manager/api"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type DaemonConfig struct {
+	CpuLimitConfig struct {
+		Enabled    bool  `json:"enabled"`
+		Limit      int64 `json:"limit,string"`
+		BurstLimit int64 `json:"burstLimit,string"`
+	} `json:"cpuLimit"`
+}
+
+func TestCpuBurst(t *testing.T) {
+	f := features.New("cpulimiting").WithLabel("component", "ws-manager").Assess("check cpu limiting", func(_ context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
```

Comment:
```suggestion
t.Parallel()

```

Reply:
It should be enough, but I increased it just in case.